plugins/woocommerce/includes/blocks/class-wc-blocks-utils.php

```diff
@@ -59,6 +59,27 @@ function ( $block ) use ( $block_name ) {
 		);
 	}
 
+	/**
+	 * Get all instances of the specified block on a specific template part.
+	 *
+	 * @param string $block_name The name (id) of a block, e.g. `woocommerce/mini-cart`.
+	 * @param string $template_part_slug The woo page to search, e.g. `header`.
+	 * @return array Array of blocks as returned by parse_blocks().
+	 */
+	public static function get_block_from_template_part( $block_name, $template_part_slug ) {
+		$template = get_block_template( get_stylesheet() . '//' . $template_part_slug, 'wp_template_part' );
+		$blocks   = parse_blocks( $template->content );
+
+		return array_values(
+			array_filter(
+				$blocks,
+				function ( $block ) use ( $block_name ) {
+					return ( $block_name === $block['blockName'] );
+				}
+			)
+		);
```

Comment:
I think that currently only works if the Mini Cart is a direct child of the Header template part, but if it's a descendant of an inner block, it doesn't. Ie: if you add the Mini Cart block inside a Group block, `mini_cart_used` is always false.

<img src="https://user-images.githubusercontent.com/3616980/165336296-9d8d58de-8fbc-44b1-89de-31b803c7d283.png" alt="" width="237" />

Reply:
Great catch! Thanks!

In [9fdc445](https://github.com/woocommerce/woocommerce/pull/32777/commits/9fdc4454cc2a1b37bcb0019aea2ff9ecb7972e7f) I created another function that created a flatten array with inner blocks too. 
There is a[ similar function in WordPress](https://developer.wordpress.org/reference/functions/_flatten_blocks/), but it seems that it is private and it was introduced only on WordPress 5.9. What should we do?

Reply:
Good finding! If it's marked as internal, I would avoid using it.